sympy/concrete/summations.py

```diff
@@ -1137,9 +1137,16 @@ def eval_sum_symbolic(f, limits):
         if without_i != 0:
             s = eval_sum_symbolic(with_i, (i, a, b))
             if s:
-                r = without_i*(b - a + 1) + s
-                if r is not S.NaN:
-                    return r
+                if((s == S.NegativeInfinity and (b - a + 1) == S.Infinity) or (s == S.Infinity and (b - a + 1) == S.Infinity)):
```

Comment:
did you mean the last S.Infinity to be S.NegativeInfinity?  If so, you could write `if (b - a - s).is_infinite:`

Reply:
An option (in case the above is not what you want to to) is to do the comparison like `s is S.Infinity` or `s in (S.Infinity, S. NegativeInfinity)` which is slightly more efficient.

Reply:
yupp cool will get this done . the Second check  by the way is for these cases where s would be returning `oo` but still won't  be dominating the other term as eventually returns `-oo` .These cases won't be addressed without the second one 
```
>>> Sum(1/x - 3, (x, 1, oo)).doit()
-oo
```